public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.ts

```diff
@@ -37,21 +44,23 @@ export class VariableSupport extends CustomVariableSupport<DataSource, AzureMoni
               data: res?.length ? [toDataFrame(res)] : [],
             };
           case AzureQueryType.ResourceGroupsQuery:
-            if (queryObj.subscription) {
+            if (queryObj.subscription && this.hasValue(queryObj.subscription)) {
               const rgs = await this.datasource.getResourceGroups(queryObj.subscription);
               return {
                 data: rgs?.length ? [toDataFrame(rgs)] : [],
               };
             }
+            return { data: [] };
```

Comment:
This was the fun bug. If a query didn't success the check (e.g. empty `queryObj.subscription`, there was no `break` in this `case` or a default `return`, that means it executed the following `case` (`NamespacesQuery`) and so on.

In the tests, legacy queries that were migrated to the new format, would fail this check (and all the checks), until eventually arrive to the `GrafanaTemplateVariableFn` case. There it executed the function in the legacy format which was mocked so it didn't even verify that the parameters were correct.

Reply:
😆 great work identifying this issue and ensuring the test cases are valid!